Lib/test/test_pkgutil.py

```diff
@@ -92,6 +93,45 @@ def test_getdata_zipfile(self):
 
         del sys.modules[pkg]
 
+    def test_issue44061_iter_modules(self):
+        #see: issue44061
+        zip = 'test_getdata_zipfile.zip'
+        pkg = 'test_getdata_zipfile'
+
+        # Include a LF and a CRLF, to test that binary data is read back
+        RESOURCE_DATA = b'Hello, world!\nSecond line\r\nThird line'
+
+        # Make a package with some resources
+        zip_file = os.path.join(self.dirname, zip)
+        z = zipfile.ZipFile(zip_file, 'w')
+
+        # Empty init.py
+        z.writestr(pkg + '/__init__.py', "")
+        # Resource files, res.txt
+        z.writestr(pkg + '/res.txt', RESOURCE_DATA)
+        z.close()
+
+        try:
+            # Check we can read the resources
+            sys.path.insert(0, zip_file)
```

Comment:
```suggestion
        # Check we can read the resources
        sys.path.insert(0, zip_file)
        try:
```

In case the insert itself fails - we don't want to clean up `sys.path[0]` in that case.